File: cpp/questions/setZeros.cpp
#include <vector>

using namespace std;

class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int rs = matrix.size();
        int cs = matrix[0].size();
        bool r00 = false;
        for (int c=0; c<cs; ++c) {
            if (matrix[0][c] == 0) {
                r00 = true;
            }
        }
        for (int r=1; r<rs; ++r) {
            bool r0 = false;
            for (int c=0; c<cs; ++c) {
                if (matrix[r][c] == 0) {
                    matrix[0][c] = 0;
                    r0 = true;
                }
            }
            if (r0) {
                std::fill(matrix[r].begin(), matrix[r].end(), 0);
            }
        }
        for (int c=0; c<cs; ++c) {
            if (matrix[0][c] == 0) {
                for (int r=1; r<rs; ++r) {
                    matrix[r][c] = 0;
                }
            }
        }
        if (r00) {
            std::fill(matrix[0].begin(), matrix[0].end(), 0);
        }
    }
};

File: python/questions/setZeros.py
from typing import List

class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        rs = len(matrix)
        cs = len(matrix[0])
        r00 = False
        for c in range(cs):
            if matrix[0][c] == 0:
                r00 = True
        for r in range(1, rs):
            r0 = False
            for c in range(cs):
                if matrix[r][c] == 0:
                    matrix[0][c] = 0
                    r0 = True
            if r0:
                for c in range(cs):
                    matrix[r][c] = 0
        for c in range(cs):
            if matrix[0][c] == 0:
                for r in range(1, rs):
                    matrix[r][c] = 0
        if r00:
            for c in range(cs):
                matrix[0][c] = 0
